provider/infoblox/infoblox_test.go

```diff
@@ -393,50 +386,112 @@ func newInfobloxProvider(domainFilter endpoint.DomainFilter, zoneIDFilter provid
+				},
+			},
+			domainFilter: endpoint.NewDomainFilter([]string{"10.0.0.0/24"}),
+			createPtr:    true,
+			expectedEndpoints: []*endpoint.Endpoint{
+				endpoint.NewEndpoint("example.com", endpoint.RecordTypePTR, "10.0.0.1"),
+				endpoint.NewEndpoint("example2.com", endpoint.RecordTypePTR, "10.0.0.2"),
+
+				// A scafold for the test case to pass,
+				// because the test framework lacks some abilities so far :-( ...
+				endpoint.NewEndpoint("example.com", endpoint.RecordTypePTR, "10.0.0.1"),
+				endpoint.NewEndpoint("example2.com", endpoint.RecordTypePTR, "10.0.0.2"),
+			},
+		},
+		// Reverse-mapping zone exists, but PTR record doesn't match the domain filter.
+		"NoPTRRecordsMatchDomainFilter": {
+			client: mockIBConnector{
+				mockInfobloxZones: &[]ibclient.ZoneAuth{
+					createMockInfobloxZone("10.0.0.0/24"),
+				},
+				mockInfobloxObjects: &[]ibclient.IBObject{
+					createMockInfobloxObject("example.com", endpoint.RecordTypePTR, "10.0.0.1"),
+				},
+			},
+			domainFilter:      endpoint.NewDomainFilter([]string{"20.0.0.0/24"}),
+			createPtr:         true,
+			expectedEndpoints: []*endpoint.Endpoint{},
 		},
 	}
 
-	providerCfg := newInfobloxProvider(endpoint.NewDomainFilter([]string{"example.com"}), provider.NewZoneIDFilter([]string{""}), true, false, &client)
-	actual, err := providerCfg.Records(context.Background())
-	if err != nil {
-		t.Fatal(err)
-	}
-	expected := []*endpoint.Endpoint{
-		endpoint.NewEndpoint("example.com", endpoint.RecordTypeA, "123.123.123.122"),
-		endpoint.NewEndpoint("example.com", endpoint.RecordTypeTXT, "\"heritage=external-dns,external-dns/owner=default\""),
-		endpoint.NewEndpoint("nginx.example.com", endpoint.RecordTypeA, "123.123.123.123"),
-		endpoint.NewEndpoint("nginx.example.com", endpoint.RecordTypeTXT, "\"heritage=external-dns,external-dns/owner=default\""),
-		endpoint.NewEndpoint("whitespace.example.com", endpoint.RecordTypeA, "123.123.123.124"),
-		endpoint.NewEndpoint("whitespace.example.com", endpoint.RecordTypeTXT, "\"heritage=external-dns,external-dns/owner=white space\""),
-		endpoint.NewEndpoint("hack.example.com", endpoint.RecordTypeCNAME, "cerberus.infoblox.com"),
-		endpoint.NewEndpoint("multiple.example.com", endpoint.RecordTypeA, "123.123.123.122", "123.123.123.121"),
-		endpoint.NewEndpoint("multiple.example.com", endpoint.RecordTypeTXT, "\"heritage=external-dns,external-dns/owner=default\""),
-		endpoint.NewEndpoint("existing.example.com", endpoint.RecordTypeA, "124.1.1.1", "124.1.1.2"),
-		endpoint.NewEndpoint("existing.example.com", endpoint.RecordTypeTXT, "\"heritage=external-dns,external-dns/owner=existing\""),
-		endpoint.NewEndpoint("host.example.com", endpoint.RecordTypeA, "125.1.1.1"),
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			providerCfg := &ProviderConfig{
+				domainFilter: tc.domainFilter,
+				zoneIDFilter: provider.NewZoneIDFilter([]string{""}),
```

Comment:
Would you mind adding test case(s) for the zone id filtering too? I know some customers do use the filtering by zone ID, it would be nice to cover this case too.